hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AsyncFSWAL.java

```diff
@@ -487,6 +490,7 @@ private void drainNonMarkerEditsAndFailSyncs() {
     while (iter.hasNext()) {
       FSWALEntry entry = iter.next();
       if (!entry.getEdit().isMetaEdit()) {
+        entry.getRpcCall().ifPresent(ServerCall::releaseByWAL);
```

Comment:
In theory,  once we remove a FSWALEntry from the unackedAppended deque,  we should release the BB backend the WAL ?  Could we define a deque subclass, once we remove the entry from it then will release it , then we don't need to release the wal everywhere, I think...

Reply:
No, if there are errors when syncing, we will put all the entries in unackedAppends back to toWriteAppends, and there we should not release the entries.